alpha branch merge to sync it with baseline

PepeAmpere opened the pull request:
notAlab alpha branch merge, bringing
- new UI
- strategic dominance end game rule as default (removing the old custom rules which were not used)
- adding support for notAmodules (base libraries, new UI, strongpoints, customCommands)
- refactorings (e.g. removing hardcoded spaces from buttons captions)
- BETS AI integration

frju365 commented on modinfo.lua:
```diff
@@ -1,12 +1,12 @@
+	shortGame		= 'ss44',
+	mutator			= 'notAlab distribution',
+	description		= 'Epic World War II RTS',
+	url				= 'http://notalab.com/products/ss44/',
```
The URL is spring1944.net not this URL... :/// And the Mutator needs to be correct too. And it's not Steam Version, and No, it's not ss44, and no, it's not an alpha-*version. ;) Need to be correct !

PepeAmpere replied:
Those complains are invalid, because this file changes won't merge anywhere. 

This is one of the manual changes which I have to do during the merge to avoid propagation of alpha setup to master branch. In previous merge this was not present

PepeAmpere replied:
Btw check this file is in conflicted state in global overview (+check what is last commit in master branch). Both because of this.

sanguinariojoe replied:
Well, that's a PR... If something shall not be merged, then you should not include it in the PR (by means of a branch, right?)

PepeAmpere replied:
Maybe I'm noob in using the github UI. I always handle `modinfo.lua` manually while propagating changes from master > alpha > beta > release branch, because each branch have different game name in given branch for lobby distinction purposes. 

In ideal world alpha branch wont be needed after this merge, so from now only master > beta > release will stay for purpose of Steam release. We will see, until its clear I will keep current modinfo.lua in alpha branch in current state. 

As I said once you give me green for finishing the PR, i will manage this file manually anyway.

sanguinariojoe replied:
I prefer to do an usual PR leaded merge process... But don't worry... Just go ahead with the other topics

PepeAmpere replied:
Ok, all topics I could solved (except potential removing of 2 old widgets - but those are not blockers for merge for me) are solved.

Waiting for your 2 TODOs. What would help me is setting some scope of definition of done of those TODOs. Maybe I can help with them or at least I will have idea how much time it will take (so I can focus on different things).

sanguinariojoe replied:
We have the license issue...

PepeAmpere replied:
addressed in [03b32ce] - more in other comments

frju365 commented on LuaRules/Gadgets/customCommands.lua:
```diff
@@ -0,0 +1,55 @@
+	desc = "Gadget connecting custom commands module allowing players to define their own commands",
+	author = "PepeAmpere",
+	date = "2017-01-24",
+	license = "notAlicense",
```
If it's notAlicense, you need to add a License file. But I think to avoid problem inside the group, you could change it to GPLv2.

PepeAmpere replied:
solved in [43c5d5f]

frju365 commented on LuaRules/Gadgets/game_strategicDominance.lua:
```diff
@@ -0,0 +1,266 @@
+	desc = "Custom endgame rules.",
+	author = "PepeAmpere",
+	date = "2018-01-29",
+	license = "notAlicense",
```
Same here

PepeAmpere replied:
solved in [43c5d5f]

frju365 commented on modules/notAchili/ss44UI/skin.lua:
```diff
@@ -0,0 +1,190 @@
+local skin = {
+	info = {
+		name = "ss44UI",
+		version = "0.1",
```
Can it be s44UI just ?

frju365 commented:
Is the  LuaUI/Fonts/visitor.txt file needed  ?

sanguinariojoe commented on LuaRules/Gadgets/game_flagManager.lua:
```diff
@@ -264,6 +264,16 @@ function PlaceFlag(spot, flagType, unitID)
+		z = spot.z,
+		radius = flagTypeData[flagType].radius,
+		income = spot.initialProduction,
+		ownerTeamID = Spring.GetGaiaTeamID(),
```
GAIA_TEAM_ID

PepeAmpere replied:
fixed in [eef5511]

sanguinariojoe commented on LuaUI/Widgets/cmd_distIcon.lua:
```diff
@@ -1,22 +0,0 @@
-
-function widget:Initialize()
-	unitIconDist = Spring.GetConfigInt('UnitIconDist')
-	Spring.SendCommands("disticon 250")
```
I see you are now dealing with the icon distance as a GUI option. I like that!
The problem I see is that if you play another game, and `UnitIconDist` is changed, then s44 parameter will be affected as well, right?

PepeAmpere replied:
If I understand the case you talk about, I think this does not happen. UIs are saving the configuration in `SpringData/LuaUI/Configs/<gameShortName>.lua` so the configuration is shared only across multiple versions of same game, not across multiple games. 

New UI is setting the default on 100 I think and then loads it from config. You can manually set your own value in `menu > settings > view > icon distance (slider)` and your custom value will be used from that moment for any other match.

This is my intention, I tested it it works this way in the most common user paths, but I admit it was not tested on all edgecases (like lobby override by starting param, other custom widgets...).

Also the current default is set roughly, just to have something smaller than 250. All the UI tweaks should be done in next iteration already in baseline. In change you comment I just removed the biggest blocker, that widget override setting it to crazy high value.

Btw regarding the default - whats good? Our experience from nota is 80-110, depends on player - and nota have still bigger avg size of unit. So I think while tweaking it for s44, going under 100 should be no problem for s44. From my experience on tradeshows people considered the icon view as the cool feature of whole game (combined with unlimited zoom-out) so this feedback supports us in this direction. Also I personally see no point to show model of the unit in distance from which the type of the unit is not visually clear. Anything above 100 is such situation.

sanguinariojoe replied:
> If I understand the case you talk about, I think this does not happen. UIs are saving the configuration in SpringData/LuaUI/Configs/<gameShortName>.lua so the configuration is shared only across multiple versions of same game, not across multiple games.

Are you positive about that? Because i detect the following in epicmenu:

```
LuaUI/Widgets/gui_notAchili_epicmenu.lua:                                       Spring.SetConfigInt( option.springsetting, BoolToInt(option.value) )
LuaUI/Widgets/gui_notAchili_epicmenu.lua:                                               Spring.SetConfigInt( option.springsetting, option.value )
```

TODO: Check it is not overriding global settings

> Btw regarding the default - whats good? Our experience from nota is 80-110, depends on player - and nota have still bigger avg size of unit. So I think while tweaking it for s44, going under 100 should be no problem for s44. From my experience on tradeshows people considered the icon view as the cool feature of whole game (combined with unlimited zoom-out) so this feedback supports us in this direction. Also I personally see no point to show model of the unit in distance from which the type of the unit is not visually clear. Anything above 100 is such situation for Infantry and there it matters the most imo.

No idea at all... I actually just edit it to make snapshots and so on... Anyway, it can be tweaked later...

PepeAmpere replied:
Commands you mentioned does not affect my global settings in lobby, but maybe my lobby is using own way to defend its global settings :)

I have not experienced any transfer of the settings while switching between multiple games. I would keep it open and react based on players feedback. The question is complex anyway - I mean its hard to prevent all possible problems while you and me are using just limited subset of lobby setups, lobby types, autohost setups and personal settings combinations possible while adding such huge new thing.

Do you agree to solve it later if anyone raise the issue?

sanguinariojoe replied:
I'm agree!

sanguinariojoe commented on LuaUI/Widgets/gui_buildbar.lua:
```diff
@@ -973,8 +973,8 @@ function widget:GetTooltip(x,y)
-             GreyStr .. "Metal cost " .. OrangeStr .. unitDef.metalCost .. "\n" ..
-             GreyStr .. "Energy cost " .. YellowStr .. unitDef.energyCost .. GreyStr .. " Build time "  .. BlueStr .. unitDef.buildTime
+             GreyStr .. "Command cost " .. OrangeStr .. unitDef.metalCost .. "\n" ..
+             GreyStr .. "Supplies cost " .. YellowStr .. unitDef.energyCost .. GreyStr .. " Build time "  .. BlueStr .. unitDef.buildTime
```
Ammo? Ammunition?

PepeAmpere replied:
Supplies => Ammunition changed in [c9b9376]

sanguinariojoe commented on LuaUI/Widgets/gui_chili_proconsole2.lua:
```diff
@@ -10,7 +10,7 @@ function widget:GetInfo()
 		layer        = 50,
 		experimental = false,
-		enabled      = true,
+		enabled      = false,
```
TODO: Check the new console

PepeAmpere replied:
What is definition of done of this TODO? 😈 Maybe I can assist somehow by answering some questions.

sanguinariojoe replied:
Just take a look... I even focused on that last time I played your UI

sanguinariojoe commented on LuaUI/Widgets/gui_notAchili_widgetselector.lua:
```diff
@@ -0,0 +1,380 @@
+function widget:GetInfo()
```
Last time it was broken... Now is OK, right?

PepeAmpere replied:
I'm not sure what last time you mean (the first versions two weeks ago? - if you mean that, it was addressed in [b3e92fb])

sanguinariojoe replied:
I pressume you fixed that then... Anyway, during the tests I'll probably check

sanguinariojoe replied:
Checked!

sanguinariojoe commented on LuaUI/Widgets/gui_chili_proconsole2.lua:
```diff
@@ -10,7 +10,7 @@ function widget:GetInfo()
 		layer        = 50,
 		experimental = false,
-		enabled      = true,
+		enabled      = false,
```
What if someone has it already enabled??
May we just remove this widget?

PepeAmpere replied:
If you support the idea, we can (i like where there is just one thing). I kept it there for people who may want to stick with old UI and maybe using some its features (i was not scanning it much).

Widgets loading is based on the config I mentioned in comment - same rules are applied here, player can customize on his side, so he will be using this console instead of the new default one.

We can remove it in the future, not blocking the merge I think.

sanguinariojoe replied:
For me you can remove that... You are replacing it with something better, right?

sanguinariojoe replied:
I changed my mind... The current console has 2 killer features:

1.- Chat log... You know? a scrollbar to check old messages
2.- When someone add a mark, an icon is added to the console message. Such icon can be used to target the camera to the mark

Probably you can easily import both features to your console

PepeAmpere replied:
* @sanguinariojoe old messages can be checked even in new console vie SHIFT+ESC
* marks are shown in minimap - I understand it may be useful for some people but e.g. I never used it, it was not clear this feature was present in the old one. Also spectators can use the stalking widget to see all actions of some players. Is it really worth of delaying the big thing by this small thing? Keeping the old solution for some time until the replacement is done in the new one can be compromise for few weeks if there are people who would really miss that feature. For majority of others it is not important I believe.

sanguinariojoe replied:
We can wait, but in such a case, don't remove the old console... It'll act as a reminder :-p

PepeAmpere replied:
Ok, keep it

sanguinariojoe commented on LuaUI/Widgets/gui_s44Only.lua:
```diff
@@ -6,7 +6,7 @@ function widget:GetInfo()
 		license = "GNU GPL v2",
 		layer = 1,
-		enabled = true,
+		enabled = false,
```
What if someone has it already enabled??
May we just remove this widget?

PepeAmpere replied:
I would be for removing this widget while there is similar logic which can be potentially used in `\LuaUI\Widgets\gui_notAchili_ss44_ui.lua` (at the bottom) which is less restrictive, killing only the most used independently distributed widgets which some legacy players can have stored in their global `SpringData/LuaUI/Widgets` folder

sanguinariojoe replied:
IDEM... If it becomes useless... For me you can remove

PepeAmpere replied:
ok, removed in  [a0b5973]

sanguinariojoe commented on modules/notAchili/ss44UI/config/tip_conf.lua:
```diff
@@ -0,0 +1,15 @@
+local ignoreweapon =
+{
+	armaak = {1},
```
There is not such weapons... or any reference to these names...

PepeAmpere replied:
removed in [e1161ae]

PepeAmpere commented on .gitmodules:
```diff
@@ -0,0 +1,15 @@
+	url = https://bitbucket.org/notabots/customcommands.git
+[submodule "modules/notAchili/data"]
+	path = modules/notAchili/data
+	url = https://bitbucket.org/notabots/notachili.git
```
Joe, regarding the investigation - what is the scope or the method? maybe i can provide data myself?

During experiments (while refactoring) I had game running with 4 instances of chili at the same time and i have not reached any gap. In profiling there were no significat spikes. Is your RAM fear based on some real experience?

Chili objects can consume few KBs and images of the UI few MBs at the most. (whole folder have 703 KB) :)

sanguinariojoe replied:
Don't worry. I'll just check we are not in a critical situation. In such a case, we can merge and I'll further investigate (eventaully opening issues or additional PRs)

PepeAmpere replied:
ok, thank you.

sanguinariojoe replied:
OK, it seems we are safe! the GUI does not seems to blow up the LUA memory over 200MB. Another question to consider in the future is where the hell is going the extra consumed memory? :-p

PepeAmpere replied:
Good, regarding this topic I understand it as a green light. I never observed huge memory consumption unless someone did some procedural magic which got out from expected complexity/size bounds. Or potentially some lame widgets using chili can create not infinit but growing recursion on chili objects (e.g. on each click instead of updating parent you create new child object).

sanguinariojoe replied:
The memory issue is closed. We can further investigate after integration. But is not a green light, we have the license issues

PepeAmpere commented on .gitmodules:
```diff
@@ -0,0 +1,15 @@
+[submodule "modules/core"]
+	path = modules/core
+	url = https://bitbucket.org/notabots/core.git
```
I was considering the mirror but I had no rights to create mirror repo here + it would be extra work which is practically not as necessary because:

1) You can do it anytime in the future from any local checkout in case of any issue.
2) Service failure of bitbucket is as same probable as service failure of github.
3) I can give anyone all rights you need on bitbucket.
4) Maintenance of mirrors adds extra cost to any update.

So its valid idea, but extra work with no practical benefit now.

-----

missing licenses were added in [43c5d5f]

sanguinariojoe replied:
OK

PepeAmpere commented on LuaRules/Configs/commandsIDs.lua:
```diff
@@ -0,0 +1,26 @@
+
+-- !OBSOLETE, keep just for backwards compatibility
+
+CMD_BUILD = 10010
```
Why here: command IDs are game rules, belongs here, not in UI module, UI module just uses them
Why some strange commands: you are right, no reason for them, copy paste nota-legacy junk: removed in [f5b3611]

Purpose of this file: 
1) now primarly documentation of dedicated ranges
2) only two remaining ones are there for missing gadgets which were planned in second wave of improvements. The reason why I dont add them once I add the gadgets is the not straightforward dependency on UI which is already active.

PepeAmpere commented on LuaRules/Gadgets/customCommands.lua:
```diff
@@ -0,0 +1,55 @@
+
+-- msg updates
+function gadget:RecvLuaMsg(msg, playerID)
+	message.Receive(msg, playerID) -- using messageReceiver data structure
```
Nice topic, thanks for linking it.

I'm aware of serialization not being much effective from performance/memory consumption perspective. Anywhere it is used, it is used for reasonable tasks :) = we send tables of low complexity never transferring e.g. huge objects (e.g. chili objects with huge metatables). In all cases its add hoc constructed table, usually 1-2 levels deep. 

Even the support for Vec3 and HMSF (time vector) I added into message system (and its [serialization and deserialization logic](https://bitbucket.org/notabots/core/commits/54457cd75e1ee45283b4ae117c92cf518b8e4216)) lately wont transfer any methods or metatables in message. E.g. Vector is translated into simple table with special flag telling the deserialization logic to create vector instead of simple table on the other side of the tunnel.

The key is to transfer the bare minimum. If sender and receiver both know the vector, i dont need to send the object properties via the tunnel ;).

sanguinariojoe commented on docs/notAlicense.txt:
```diff
@@ -0,0 +1,19 @@
+
+Software and associated documentation files (the "Software") published under this license is originally released under MIT license (see below).
+
+notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
```
That's in fact GPL incompatible... See:

https://en.wikipedia.org/wiki/License_compatibility#GPL_compatibility

MIT license is GPL compatible (in the context of MIT code embedded in GPL code) precisely because MIT code is then overridden.

In other words, notAlicense is actually "not a license for GPL projects", which is specifically forbidding the usage of your code inside GPL software.

How to fix that? Super-easy! Just relicense all your codes as MIT... Then, we are fulfilling everything:

1.- We are keeping the original copyright note, the very only restriction of MIT
2.- Out license states that code taken from this repo is GPL, regardless of the original license.

PepeAmpere replied:
This is exactly what is notAlicense saying :) The life of the code from now is under whatever global license the product is using.  Ín license I just explicitly say that original version of my code is MIT so I can freely distribute that (and only that) version of code under MIT. Any derivative work which is produced by anyone else in s44 repository will use that whatever global license with all consequences - in this case GPL vN.

specing replied:
Then why not use MIT directly? It is GPL compatible: https://www.gnu.org/licenses/license-list.en.html#X11License

Your notalicense just adds confusion and more text to the repo.

PepeAmpere replied:
notAlicense is solving the potential uncertainty regarding the first public publishing of the code in globally GPL environment - so in fact it is reducing the confusion. 

I hope "more text to the repo" was not intended as a serious argument.

sanguinariojoe replied:
No Pepe, is not... The origin of the software is protected by MIT license, which states that the original copyright notice shall be preserved.

Your text is making the license GPL incompatible. You are spoecifically saying that "Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.", and indeed if we are not modifying the software then we cannot redistribute it as GPL... You see the problem?

PepeAmpere replied:
1) I make original version of the software Xv1. 
2) Publish it Xv1 as notAlicense(MIT) on my repo.
3) I publish Xv1 as notAlicense(MIT) on s44 repo.
4) From that moment Xv1 is GPL positive on your repo because thats the licensing scheme of your distribution, you are allowed to distribute it under GPL from the beginning.
5) I just say that Xv1 storred on my repo is still MIT no matter what happens with X in s44 repo.

frju365 replied:
@PepeAmpere : you have a strange vision of licensing software :-D Contact the FSF but I'm not sure it will agree with you.

btw: saying : notAlicense(MIT) is not allowed by the MIT license. To really be GPL compatible and to not break the rules set by the MIT, you need to have just this text : 
```
Copyright 2018 NotaLab, Sping1944

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

Other are not allowed.

frju365 replied:
Licencing software is not a game btw... it's all the more serious we are speaking about free softwares.  Renaming a licence make it proprietary and not-GPL-Compatible. The ONLY thing written where the name of the orga can appear is the first line : 
> Copyright (C) \<date\> \<project\> 

But you can't call this NotaLicense is not really allowed , in your work or in this repo. It's forbidden.

example of an MIT license from a company : https://github.com/roadiz/roadiz/blob/master/LICENSE.md

ashdnazg replied:
No notalicense in spring1944 please. @FLOZi @kanatohodets

sanguinariojoe replied:
Pepe, again, you are not saying that:

> ... does not restrict any use of the **original version of the Software** ...

And again, the MIT license already make the cover you are trying to get.

Of course, I concern here about the integration in S44. Said that, as a friend I would warn you that your license is even MIT incompatible. Personally I guess your license changes don't worth eventual legal issues...

We can discuss further if you want, but I'll complain as many as required to avoid getting integrated software under notAlicense terms. The last thing I want in my life is a lawyer threatening me

kanatohodets replied:
Wow, busy PR! Yes, please no custom licenses. [Spring's stance](https://springrts.com/phpbb/viewtopic.php?f=1&t=17847) is pretty clear: Spring-related Lua code is considered linking against the engine, and so those works must be GPL, full-stop. I think the simplest thing is for _any_ Spring Lua code to be GPL licensed from the outset; any more concrete recommendation than that starts to get into long hours of reading license compatibility write-ups and playing lawyer.

From a practical standpoint using custom variations of well-known licenses is likely to get the work excluded from free software repositories: Debian, for example, just doesn't have time or legal risk tolerance to closely read your license and confirm its legal integrity. There's a can of worms regarding these repositories and the CC-BY-NC license that the art is under, but there's no good reason to make that situation worse.

In terms of what the code was originally published as -- this doesn't matter if you're the sole copyright holder. You can just contribute it to this project under the terms of an entirely different license.

sanguinariojoe replied:
@PepeAmpere has already removed the notAlicense reference in the internal code, but the submodules are still licensed under notAlicense GPL incompatible terms. Those codes are in fact downloaded and deployed within the game each time you log in the host, so a license collision remains.

PepeAmpere commented:
@sanguinariojoe hm, i tried to resolve the conflicts of `modinfo.lua` just for review purposes but only confirm button is saying "merge" which I suspect from finishing the Pull/Merge request :) I would like to prevent the impression I merge something without you finishing the review so I can experiment later with github once your part is done. Currently waiting for:

* 2 TODOs (RAM and console related)
* confirmation for removing two widgets which have replacements in new UI
* closing the licensing discussion

sanguinariojoe commented on LuaRules/Configs/commandsIDs.lua:
```diff
@@ -0,0 +1,14 @@
+--  Proposed Command ID Ranges:
```
Please, move this file to your modules folder (it is indeed used just there)

PepeAmpere replied:
I'm not sure if you read my previous reaction on this. Once I edited the lines, the comment is lost :)

sanguinariojoe replied:
Yeah, i didn't, sorry! Would you repeat it to me?

PepeAmpere replied:
Ok, this time I will be a bit shorter:

1) this file is setting rules, setting global constants. Such things have to be in LuaRules (or gamedata), not in widgets/modules, because cmdIDs is something other gadgets/widgets would need to know/share.
2) you are right current version is linked only from new UI. Two reasons behind it: 
2a) Originally I planned adding scatter gadget (new command) and buildspeed gadget for factories (25%, 50%, 75%, 100% settings commands) to s44 but refactoring of those old things would be costly so I re-planned the delivery of those features for next iteration. Those gadgets would use the constants stored here.
2b) The file provides documentaiton: shows people suggested ranges of commands for any new definitions we may declare for any gadget
3) Based on 2a it would make sense to remove this file completely once those commands do not practically exist but I kept it here because: temporary cutting the dependent logic in UI is not trivial + if someone would pick those cmdIDs in next weeks, it would be another thing to solve while I will port the gadgets.

So - I cleaned the file so it does not contain all other junk commands as you suggested in  your last deleted comment and keep just the necessary minimum - two commands + docu - to use the easiest way to go forward

sanguinariojoe replied:
Fair enough to me

sanguinariojoe commented:
BUG: Open Settings, press back, and a useless '...' button will be shown to you.

In first instance, why are you allowed to press back on settings?

sanguinariojoe commented:
BUG: Settings/Interface/Interface skin...

I clicked both 'default s44' and 'carbon', but nothing happened... I pressed back and the interface become ugly... Win95 skin? :-)

It's irreversible...

sanguinariojoe commented:
BUG: Settings/View/Map/VR Grid

`[widgets.lua] Error: ToggleWidget(), could not find widget: External VR Grid`

PepeAmpere commented:
@sanguinariojoe 

* BUG: Settings/View/Map/VR Grid - fixed in [747bb85] by adding the widget
* BUG: Settings/Interface/Interface skin...  - fixed in [747bb85] by correcting the UI skin shortname reference
* BUG: Open Settings, press back, and a useless '...' button will be shown to you. - fixed in [747bb85] - gets you back to previous dialog (same as `close` in given context)
* removing mentions of notAlicense in [03b32ce] to avoid confusion, follows global project licensing scheme for this distribution implicitly. 

(all under  [https://trello.com/c/PodkNXEb/])

PepeAmpere commented:
@sanguinariojoe @kanatohodets @ashdnazg @frju365 guys seems deleting of the notAlicense mentions from the repo killed the whole discussion thread it was there.

So I explicitly write this comment to keep you informed: 
* notAlicense mentions are removed from gadgets, 
* txt with notAlicense is removed as well from the repo. 
* from the beginning I'm fine with spreading my contributions in this project under global project licensing setup

For record (while all discussion is lost) I repeat that by committing the code here in this repo I have not lost the right to distribute code 100% created by me under any other license in different project.

I hope this ends the confusion and we can move forward.

PepeAmpere commented:
@sanguinariojoe please make final review, I hope I addressed all remaining issues.

sanguinariojoe commented:
Sorry, but the license issue is not closed... it seems the submodules are still licensed with a custom GPL incompatible license...

I updated the submodules:

```
git submodule update
Submodule path 'modules/core': checked out '69e7252a5ae18de817a720fe297b37e4eac34f2a'
Submodule path 'modules/customCommands/data': checked out 'db46843dab3a7252e25f7bd894fce05d265eeeb0'
Submodule path 'modules/goals/data': checked out 'c837fc71d6318413121d11155ac88f739d6b36c7'
Submodule path 'modules/strongpoints/data': checked out 'e81176d9f4bc8be33d2dbd9b189b9a59777319e0'
```

And checked for notalicense:

```
grep -ri "notalicense"
Binary file .git/modules/modules/strongpoints/data/index matches
Binary file .git/modules/modules/core/index matches
Binary file .git/modules/modules/customCommands/data/index matches
Binary file .git/modules/modules/goals/data/index matches
modules/strongpoints/data/strongpoints.lua:     license = "notAlicense",
modules/strongpoints/data/notAlicense.txt:notAlicense
modules/strongpoints/data/notAlicense.txt:notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
modules/core/mod/attach/attach.lua:     license = "notAlicense",
modules/core/ext/tableExt/tableExt.lua: license = "notAlicense",
modules/core/ext/stringExt/stringExt.lua:       license = "notAlicense",
modules/core/ext/timeExt/timeExt.lua:   license = "notAlicense",
modules/core/api/message/message.lua:   license = "notAlicense",
modules/core/api/cmdDesc/cmdDesc.lua:   license = "notAlicense",
modules/core/vec3/vec3.lua:     license = "notAlicense",
modules/core/hmsf/hmsf.lua:     license = "notAlicense",
modules/core/tsp/actionTypes/actionTypes.lua:   license = "notAlicense",
modules/core/notAlicense.txt:notAlicense
modules/core/notAlicense.txt:notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
modules/customCommands/data/api/messageSenderInternal.lua:      license = "notAlicense",
modules/customCommands/data/api/messageReceiverSynced.lua:      license = "notAlicense",
modules/customCommands/data/api/messageReceiverUnsynced.lua:    license = "notAlicense",
modules/customCommands/data/api/messageSenderExternalWidget.lua:        license = "notAlicense",
modules/customCommands/data/customCommands.lua: license = "notAlicense",
modules/customCommands/data/notAlicense.txt:notAlicense
modules/customCommands/data/notAlicense.txt:notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
modules/goals/data/goals.lua:   license = "notAlicense",
modules/goals/data/notAlicense.txt:notAlicense
modules/goals/data/notAlicense.txt:notAlicense allows embedding of the Software into GNU GPL licensed projects. Publishing of the Software in such projects does not restrict any use of the original version of the Software under MIT license.
```

PepeAmpere commented:
@sanguinariojoe if you have problems with updating the submodules, follow hint sequence in https://trello.com/c/8yoUDzHj/ and in step A2 just checkout the master

sanguinariojoe commented:
From my side, I unlock the merge process...

> BUG: Settings/Interface/Interface skin... - fixed in [747bb85] by correcting the UI skin shortname reference

It seems it fixed something. But anyway I noticed changes are not effective on already shown windows (like the command and ammo bars) until you reboot. Something to deal with by an issue I guess

PepeAmpere commented:
@sanguinariojoe deleting `SpringData/LuaUI/configs/ss44.lua` before game launch will solve all the issues caused by previous broken version.
